src/containers/auth/register/index.js

```diff
@@ -21,6 +22,19 @@ class Register extends Component {
     errors: {},
   };
 
+  componentDidMount() {
+    const { location, history } = this.props;
+    const parsed = queryString.parse(location.search);
+    if (parsed.tokenGoogle) {
```

Comment:
make HOC for google/linkedin login and provide as children needed inputs and button. CC @universezzz